app/modules/intelligence/agents/custom_agents/custom_agent.py

```
@@ -27,36 +27,24 @@
-            async with httpx.AsyncClient() as client:
-                response = await client.get(url, headers=headers)
-                response.raise_for_status()
-                data = response.json()
-                return data.get("system_prompt")
-        except Exception as e:
-            logger.error(f"Error fetching system prompt: {str(e)}", exc_info=True)
-            raise ValueError(f"Failed to fetch system prompt: {str(e)}")
+    async def _get_llm(self):
+        """Helper method to get or initialize LLM with caching"""
+        if self._llm is None:
+            self._llm = await self._llm_provider.get_small_llm(agent_type=AgentType.LANGCHAIN)
```

Ruff (0.8.2)

Undefined name `AgentType`

(F821)